internal/pkg/service/buffer/store/cleanup.go

```diff
@@ -0,0 +1,91 @@
+package store
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/filestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/slicestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+const BoundaryInDays = 14
+
+// Cleanup deletes all unneeded tasks, files and slices data from the store.
+func (s *Store) Cleanup(ctx context.Context, receiver model.Receiver, logger log.Logger) (err error) {
+	_, span := s.tracer.Start(ctx, "keboola.go.buffer.store.Cleanup")
+	defer telemetry.EndSpan(span, &err)
+
+	s.cleanupTasks(ctx, receiver.ReceiverKey, logger)
+
+	for _, e := range receiver.Exports {
+		s.cleanupFiles(ctx, e.ExportKey, logger)
+	}
+
+	return nil
+}
+
+func (s *Store) cleanupTasks(ctx context.Context, receiverKey key.ReceiverKey, logger log.Logger) {
+	tasks, err := s.schema.Tasks().InReceiver(receiverKey).GetAll().Do(ctx, s.client).All()
+	if err != nil {
+		logger.Error(err)
+		return
+	}
+	for _, t := range tasks {
+		if !t.Value.IsForCleanup() {
+			continue
+		}
+
+		_, err = s.schema.Tasks().ByKey(t.Value.TaskKey).Delete().Do(ctx, s.client)
+		if err != nil {
+			logger.Error(err)
+		}
+		logger.Infof(`deleting task "%s"`, t.Value.TaskKey.String())
```

Comment:
Jedna věc, kterou bude asi potřeba udělat jinak je předávání tady toho loggeru. 🙂 Když použiju `s.logger`, tak v lozích nebude `[service][cleanup]` prefix.

Reply:
Nebo to je takto ok?

Reply:
Nesedi mi celkom, ze "bussines logika" toho mazania je tu v `store`.

Skor by som cakal, ze tu bude iba `store.GetReceiverTasks -> iterator` a pod.

Ta logika by mala byt skor v `service` a tak nebudes mat problem s tym loggerom.

Ak nie v service (ze tam toto nechceme mat privela), tak kludne nejaky `service/buffer/clenaup` balicek.

Bola by tam nejaka struktura `Cleaner{logger}` a nad tym tieto metody.

Ten logger ti vlastne ukazuje, ze to sem nepatri.

Reply:
Jakub, k tomu si sa nevyjadril.

Reply:
@JakubMatejka neviem ci to teda ides nejak upravovat alebo nie, ale ak sa da, tak toto prosim mergnime, nech mozem rebasnut svoj PR a nerobit konflikty.

Reply:
Najprv asi potrebujes mergnut https://github.com/keboola/keboola-as-code/pull/1183, a tu v `task` testoch upravit `ExprotKey -> ReceiverKey`.

Reply:
Jj. Upravím, na to jsem včera zapomněl.

Reply:
Jestli teda chceš čekat. Asi to bude chvíli trvat.

Reply:
@JakubMatejka no len som chcel, aby si toto mergol, a spravil to v dalsom PR, ak sa da

Reply:
kedze tato verzia je funkcna

Reply:
Aha, ok

Reply:
Udělal jsem na to issue: https://keboola.atlassian.net/browse/PSGO-67